dev/tools/test-e2e

```diff
@@ -27,17 +93,30 @@ def main():
     artifact_dir = os.getenv("ARTIFACTS")
     if not artifact_dir:
         artifact_dir = f"{repo_root}/bin"
+    os.makedirs(artifact_dir, exist_ok=True)
 
     env = os.environ.copy()
     env["IMAGE_TAG"] = utils.get_image_tag()
 
-    result = subprocess.run(utils.go_tool_args(
-        "gotestsum",
-        f"--junitfile={repo_root}/bin/e2e-junit.xml",
-        f"--jsonfile={artifact_dir}/test-e2e.json",
-        "--", "./test/e2e/...", "--parallel=1"
-    ), env=env, cwd=repo_root)
-    return result.returncode
+    if not setup_python_sdk(repo_root, env):
+         return 1
+
+    go_test_result = run_go_e2e_tests(repo_root, artifact_dir, env)
+    python_test_result = run_python_e2e_tests(repo_root, artifact_dir, env)
+
+    if go_test_result != 0:
+        print("Go E2E tests failed.")
+
+    if python_test_result != 0:
+        print("Python SDK E2E tests failed.")
+        return 1
```

Comment:
The script will exit if the Python tests fail, and it won't report the Go test failures if both were to fail. Please adjust it to ensure all failures are reported before the script exits.